src/main/java/org/terasology/moduletestingenvironment/MTEExtension.java

```diff
@@ -31,21 +31,60 @@
 import java.util.function.Function;
 
 /**
- * Junit 5 Extension for using {@link ModuleTestingHelper} in your test.
+ * Sets up a Terasology environment for use with your {@index JUnit} 5 test.
  * <p>
- * Supports Terasology's DI as in usual Systems. You can inject Managers via {@link In} annotation, constructor's or
- * test's parameters. Also you can inject {@link ModuleTestingHelper} itself.
+ * Supports Terasology's DI as in usual Systems. You can inject Managers via {@link In} annotation, constructors or
+ * test parameters. Also you can inject {@link MainLoop} or {@link Engines} to interact with the environment's engine.
  * <p>
- * Every class annotated with this will create a single {@link ModuleTestingHelper} and use it during execution of
+ * Example:
+ * <pre><code>
+ * import org.junit.jupiter.api.extension.ExtendWith;
+ * import org.junit.jupiter.api.Test;
+ * import org.terasology.engine.registry.In;
+ *
+ * &#64;{@link org.junit.jupiter.api.extension.ExtendWith}(MTEExtension.class)
+ * &#64;{@link Dependencies}("MyModule")
+ * &#64;{@link UseWorldGenerator}("Pathfinding:pathfinder")
+ * public class ExampleTest {
+ *
+ *     &#64;In
+ *     EntityManager entityManager;
+ *
+ *     &#64;In
+ *     {@link MainLoop} mainLoop;
+ *
+ *     &#64;Test
+ *     public void testSomething() {
+ *         // …
+ *     }
+ *
+ *     // Injection is also applied to the parameters of individual tests:
+ *     &#64;Test
+ *     public void testSomething({@link Engines} engines, WorldProvider worldProvider) {
+ *         // …
+ *     }
+ * }
+ * </code></pre>
+ * <p>
+ * You can configure the environment with these additional annotations:
+ * <dl>
+ *     <dt>{@link Dependencies @Dependencies}</dt>
+ *     <dd>Specify which modules to include in the environment. Put the name of your module under test here.
+ *         Any dependencies these modules declare in <code>module.txt</code> will be pulled in as well.</dd>
+ *     <dt>{@link UseWorldGenerator @UseWorldGenerator}</dt>
+ *     <dd>The URN of the world generator to use. Defaults to {@link org.terasology.moduletestingenvironment.fixtures.DummyWorldGenerator},
+ *         a flat world.</dd>
+ * </dl>
```

Comment:
I really like this! :green_heart: 
However, I wonder whether this should be added to the README or module docs - not sure whether in addition to having it here or instead :thinking:

Reply:
It's definitely the sort of thing that anyone interested in using MTE will need to know. As for how to organize the docs, I haven't figured that out yet. Neither javadoc nor GitHub-Markdown offer good ways to include source listings like ExampleTest.java.